Compatibility: streaming/test_scalar_topic_write.py.TestScalarTopicWriteRollingUpgradeAndDowngrade.test_rolling_upgrade — DeadlineExceeded waiting for node readiness after rolling restart

## Summary

During a rolling `stable-26-2-1 → prestable-26-3` upgrade, the readiness probe after a node roll never succeeds: `execute_with_retries` in `_wait_for_readiness` times out with gRPC `DEADLINE_EXCEEDED` → `ydb.issues.DeadlineExceed`. The rolled node is subsequently SIGKILLed by the harness.

## Environment

- **Workflow:** [Regression-run_compatibility #28482735803](https://github.com/ydb-platform/ydb/actions/runs/28482735803)
- **Build preset:** `relwithdebinfo` (job `main:relwithdebinfo`, S3 path `ya-main-x86-64`)
- **Branch:** `main`
- **Workflow commit:** [`d4acace`](https://github.com/ydb-platform/ydb/commit/d4acacea1f802786a0c68b1753527181d82ddd5b)
- **Versions under test** (the test downloads pre-built binaries for these refs):
  - init `YDB_COMPAT_INIT_REF` = **stable-26-1-1** — [`bb4e61a`](https://github.com/ydb-platform/ydb/commit/bb4e61afa6a2d86e48d6eb1147e6ee9a5866045f)
  - inter `YDB_COMPAT_INTER_REF` = **stable-26-2-1** — [`900757a`](https://github.com/ydb-platform/ydb/commit/900757a65353c125d0aeaeea3d60ac16eea49944)
  - target `YDB_COMPAT_TARGET_REF` = **prestable-26-3** — [`687cfdc`](https://github.com/ydb-platform/ydb/commit/687cfdc66f53ee3c59cc703b1f73df7477bce65f)

## HTML reports (all attempts)

- [Try 1 — failures/mutes appear here](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/28482735803/ya-main-x86-64/try_1/ya-test.html#FAIL)
- [Try 2 — rerun of the failed chunk, passed](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/28482735803/ya-main-x86-64/try_2/ya-test.html)

## Affected tests (MUTE (try_1))

- `streaming/test_scalar_topic_write.py::TestScalarTopicWriteRollingUpgradeAndDowngrade::test_rolling_upgrade[rolling_stable-26-2-1_to_prestable-26-3-False]`

## Error

```
contrib/python/ydb/py3/ydb/connection.py:383: in __next__
  ...
E   grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
E   	status = StatusCode.DEADLINE_EXCEEDED
E   	details = "Deadline Exceeded"

ydb/tests/compatibility/streaming/test_scalar_topic_write.py:132: in test_rolling_upgrade
ydb/tests/library/compatibility/fixtures.py:367: in roll
ydb/tests/library/compatibility/fixtures.py:310: in _wait_for_readiness
ydb/tests/library/compatibility/fixtures.py:304: in _wait_for_readiness
contrib/python/ydb/py3/ydb/query/pool.py:259: in execute_with_retries
  ...
E   ydb.issues.DeadlineExceed: Deadline exceeded on request

stop: Exit code is not checked, cos binary was stopped by sigkill
```

## Reproduction

```bash
./ya make -tA --build relwithdebinfo \
  -DYDB_COMPAT_INIT_REF=stable-26-1-1 \
  -DYDB_COMPAT_INTER_REF=stable-26-2-1 \
  -DYDB_COMPAT_TARGET_REF=prestable-26-3 \
  ydb/tests/compatibility/streaming -F '*TestScalarTopicWriteRollingUpgradeAndDowngrade*'
```

> The three `-D…_REF` flags are mandatory — they select the pre-built binaries for the
> versions under test.

Per-node `stdout`/`stderr` (with the native crash stack, if any) are in the test artifacts
archive: `try_1/test_artifacts/ydb-tests-compatibility-streaming.zip`.

## Notes

The node under roll never becomes ready (all readiness queries time out), then is SIGKILLed — consistent with the target `prestable-26-3` node hanging/crashing on restart in the mixed cluster.

---
*Filed from acceptance triage of [Regression-run_compatibility #28482735803](https://github.com/ydb-platform/ydb/actions/runs/28482735803) (relwithdebinfo). Mute-bot issues (title starting with "Mute") and closed issues are cross-referenced only, not treated as active trackers.*

--- a/ydb/tests/library/compatibility/fixtures.py
+++ b/ydb/tests/library/compatibility/fixtures.py
@@ -298,9 +298,9 @@ class RollingUpgradeAndDowngradeFixture:
         request_timeout = 10  # seconds
         settings = (
             ydb.BaseRequestSettings()
-                .with_timeout(request_timeout)
-                .with_operation_timeout(request_timeout)
-                .with_cancel_after(request_timeout)
+            .with_timeout(request_timeout)
+            .with_operation_timeout(request_timeout)
+            .with_cancel_after(request_timeout)
         )
 
         start_time = time.time()

--- a/ydb/tests/library/compatibility/fixtures.py
+++ b/ydb/tests/library/compatibility/fixtures.py
@@ -55,7 +55,7 @@ def prepare_feature_flags(extra_feature_flags, disabled_feature_flags):
 
 
 def prepare_table_service_config(table_service_config):
-    table_service_config = copy.copy(table_service_config)
+    table_service_config = copy.copy(table_service_config or {})
 
     if "enable_compile_cache_warmup" not in table_service_config:
         table_service_config["enable_compile_cache_warmup"] = False
@@ -143,7 +143,7 @@ class RestartToAnotherVersionFixture:
             use_in_memory_pdisks=kwargs.pop("use_in_memory_pdisks", False),
             extra_feature_flags=extra_feature_flags,
             disabled_feature_flags=disabled_feature_flags,
-            table_service_config=kwargs.pop("table_service_config", {}),
+            table_service_config=prepare_table_service_config(kwargs.pop("table_service_config", {})),
             **kwargs,
         )
 
@@ -229,7 +229,7 @@ class MixedClusterFixture:
             suppress_version_check=not all_versions_numbered,
             extra_feature_flags=extra_feature_flags,
             disabled_feature_flags=disabled_feature_flags,
-            table_service_config=kwargs.pop("table_service_config", {}),
+            table_service_config=prepare_table_service_config(kwargs.pop("table_service_config", {})),
             **kwargs,
         )
 
@@ -320,7 +320,7 @@ class RollingUpgradeAndDowngradeFixture:
             use_in_memory_pdisks=kwargs.pop("use_in_memory_pdisks", False),
             extra_feature_flags=extra_feature_flags,
             disabled_feature_flags=disabled_feature_flags,
-            table_service_config=kwargs.pop("table_service_config", {}),
+            table_service_config=prepare_table_service_config(kwargs.pop("table_service_config", {})),
             **kwargs,
         )
 

--- a/ydb/tests/library/compatibility/fixtures.py
+++ b/ydb/tests/library/compatibility/fixtures.py
@@ -295,22 +295,38 @@ class RollingUpgradeAndDowngradeFixture:
         ) """
         timeout = 120  # seconds
         interval = 2  # seconds
+        request_timeout = 10  # seconds
+        settings = (
+            ydb.BaseRequestSettings()
+                .with_timeout(request_timeout)
+                .with_operation_timeout(request_timeout)
+                .with_cancel_after(request_timeout)
+        )
 
         start_time = time.time()
         last_exception = None
+        attempt = 0
         while time.time() - start_time < timeout:
+            attempt += 1
             try:
+                logger.info("Readiness check attempt %d", attempt)
                 with ydb.QuerySessionPool(self.driver) as session_pool:
-                    session_pool.execute_with_retries(query, retry_settings=ydb.RetrySettings(max_retries=1))
+                    session_pool.execute_with_retries(query, retry_settings=ydb.RetrySettings(max_retries=1), settings=settings)
                 break
             except Exception as e:
                 last_exception = e
+                logger.warning(
+                    "Readiness check attempt %d failed after %.1fs: %r",
+                    attempt,
+                    time.time() - start_time,
+                    e,
+                )
                 time.sleep(interval)
         else:
             raise last_exception
         query = """DROP TABLE `test_readiness`"""
         with ydb.QuerySessionPool(self.driver) as session_pool:
-            session_pool.execute_with_retries(query)
+            session_pool.execute_with_retries(query, settings=settings)
 
     def setup_cluster(self, tenant_db=None, **kwargs):
         extra_feature_flags, disabled_feature_flags = prepare_feature_flags(kwargs.pop("extra_feature_flags", []), kwargs.pop("disabled_feature_flags", []))

